src/main/java/iudx/resource/server/database/async/AsyncServiceImpl.java

```diff
@@ -88,48 +84,54 @@ public AsyncService asyncStatus(
             if (results.isEmpty()) {
-              responseBuilder =
-                  new ResponseBuilder("failed")
-                      .setTypeAndTitle(400)
-                      .setMessage("Fail: Incorrect search ID");
-              handler.handle(Future.failedFuture(responseBuilder.getResponse().toString()));
-            } else {
-              JsonObject answer = results.getJsonObject(0);
+              sendErrorResponse(handler, ResponseUrn.BAD_REQUEST_URN, "Fail: Incorrect search ID");
+              return;
+            }
 
-              String userId = answer.getString("user_id");
-              if (sub.equals(userId)) {
-                String status = answer.getString(STATUS);
-                if (status.equalsIgnoreCase(QueryProgress.COMPLETE.toString())) {
-                  answer.put(FILE_DOWNLOAD_URL, answer.getValue(S3_URL));
-                }
-                answer.put("searchId", answer.getString("search_id"));
-                answer.put("userId", userId);
-
-                answer.remove(S3_URL);
-                answer.remove("search_id");
-                answer.remove(USER_ID);
-                LOGGER.debug(answer.encodePrettily());
-                JsonObject response =
-                    new JsonObject()
-                        .put("type", ResponseUrn.SUCCESS_URN.getUrn())
-                        .put("title", ResponseUrn.SUCCESS_URN.getMessage())
-                        .put("results", new JsonArray().add(answer));
-                handler.handle(Future.succeededFuture(response));
-              } else {
-                responseBuilder =
-                    new ResponseBuilder("failed")
-                        .setTypeAndTitle(400, ResponseUrn.BAD_REQUEST_URN.getUrn())
-                        .setMessage(
-                            "Please use same user token to check status as "
-                                + "used while calling search API");
-                handler.handle(Future.failedFuture(responseBuilder.getResponse().toString()));
-              }
+            AsyncStatusQueryResult queryResult =
+                new AsyncStatusQueryResult(results.getJsonObject(0));
+            LOGGER.debug(queryResult.toJson());
+            LOGGER.debug(queryResult.getUserId());
+            if (!sub.equals(queryResult.getUserId())) {
+              sendErrorResponse(
+                  handler,
+                  ResponseUrn.BAD_REQUEST_URN,
+                  "Please use the same user token to check status as used while calling the search API");
+              return;
             }
+            LOGGER.debug("here");
+            processQueryResult(queryResult, handler);
           }
         });
     return this;
   }
 
+  private void processQueryResult(
+      AsyncStatusQueryResult queryResult, Handler<AsyncResult<JsonObject>> handler) {
+    if (queryResult.getStatus().equalsIgnoreCase(QueryProgress.COMPLETE.toString())) {
+      queryResult.setFileDownloadUrl(null);
```

Comment:
sorry, I couldn't understand why .setFileDownloadUrl() and .setSearchId() with null are called